pystac_client/client.py

```diff
@@ -213,21 +225,86 @@ def from_file(  # type: ignore
             )
 
         client: Client = super().from_file(href, stac_io)
-
-        client._stac_io._conformance = client.extra_fields.get(  # type: ignore
-            "conformsTo", []
-        )
         client.modifier = modifier
 
         return client
 
-    def _supports_collections(self) -> bool:
-        return self._conforms_to(ConformanceClasses.COLLECTIONS) or self._conforms_to(
-            ConformanceClasses.FEATURES
+    def has_conforms_to(self) -> bool:
+        """Whether server contains list of ``"conformsTo"`` URIs"""
+        return "conformsTo" in self.extra_fields
+
+    def get_conforms_to(self) -> List[str]:
+        """List of ``"conformsTo"`` URIs
+
+        Return:
+            List[str]: List of  URIs that the server conforms to
+        """
+        return cast(List[str], self.extra_fields.get("conformsTo", []).copy())
+
+    def set_conforms_to(self, conformance_uris: List[str]) -> None:
+        """Set list of ``"conformsTo"`` URIs
+
+        Args:
+            conformance_uris : URIs indicating what the server conforms to
+        """
+        self.extra_fields["conformsTo"] = conformance_uris
+
+    def clear_conforms_to(self) -> None:
+        """Clear list of ``"conformsTo"`` urls"""
+        self.extra_fields.pop("conformsTo", None)
+
+    def add_conforms_to(self, name: str) -> None:
+        """Add ``"conformsTo"`` by name.
+
+        Args:
+            name : name of :py:class:`ConformanceClasses` keys to add.
+        """
+        conforms_to = self.get_conforms_to()
+
+        conformance_class = getattr(ConformanceClasses, name.upper(), None)
+        if conformance_class is None:
+            raise ValueError(f"Invalid conformance class {conformance_class}")
+
+        conforms_to.append(ConformanceClasses.valid_uri(conformance_class.value))
```

Comment:
I don't think it quite works because conforms_to is a list of uris, but I'm not really worried about the performance implications of just using `self.conforms_to`